### PR TITLE
Do not use external `docker` binary for `kubermatic-installer mirror-images`

### LINKED ISSUE #9548: Make image-loader utility usable without docker
### What would you like to be added? / User Story

The `kkp-installer mirror-images` tool that we distribute to pull images used by KKP and push them to a private registry requires the `docker` CLI to be installed and uses it to run pull, retagging and push operations.

I'd like to see a "docker-less" mode of it that is capable of downloading, retagging and pushing images. There's no strict need for a real container runtime when working with images (they are glorified archive files; [`skopeo`](https://github.com/containers/skopeo) doesn't need it either) and it would make `image-loader` more portable (for example, it could be used in CI pipelines that do not have access to a docker daemon).

https://github.com/containers/image might be a library worth exploring.

### Acceptance criteria

- `kkp-installer mirror-images` has a "docker-less" mode that pulls and pushes images without using the `docker` command.

### PR DESCRIPTION
**Pull request by embik**
**What this PR does / why we need it**:
This PR removes the logic that "shells out" to `docker` (or a compatible CLI) and instead uses https://github.com/google/go-containerregistry for what the `mirror-images` subcommand does. It removes `--docker-binary` as a flag since it is not needed anymore.

For credentials, `mirror-images` will automatically try to use docker's configuration file for credentials (`~/.docker/config.json`). This can be overridden by setting the `DOCKER_CONFIG` environment variable (mind that this points to a _directory_, not a file; so `DOCKER_CONFIG=/path/to/my/directory` would be right for a configuration file sitting at `/path/to/my/directory/config.json`).

**Which issue(s) this PR fixes**:

Fixes #9548

**What type of PR is this?**
/kind feature

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
Remove dependency on `docker` binary when using `kubermatic-installer mirror-images` (removes the `--docker-binary` flag)
```

**Documentation**:

```documentation
https://github.com/kubermatic/docs/pull/1307
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -349,3 +349,14 @@ require (
 	sigs.k8s.io/kustomize/kyaml v0.13.9 // indirect
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
 )
+
+require github.com/moby/sys/mountinfo v0.6.2 // indirect
+
+require github.com/google/go-containerregistry v0.12.1
```
Comment (xmudrii) [inline go.mod:355]: Why this line ended up here? I'd expect it with other imports. Have you manually added it?

File: cmd/kubermatic-installer/cmd_mirror_images.go [hunk 2]
```diff
@@ -104,8 +101,6 @@ func MirrorImagesCommand(logger *logrus.Logger, versions kubermaticversion.Versi
 	cmd.PersistentFlags().StringVar(&opt.HelmValuesFile, "helm-values", "", "Use this values.yaml when rendering Helm charts")
 	cmd.PersistentFlags().StringVar(&opt.HelmBinary, "helm-binary", opt.HelmBinary, "Helm 3.x binary to use for rendering charts")
 
-	cmd.PersistentFlags().StringVar(&opt.DockerBinary, "docker-binary", opt.DockerBinary, "docker CLI compatible binary to use for pulling and pushing images")
```
Comment (xmudrii) [inline cmd/kubermatic-installer/cmd_mirror_images.go:107]: I'm wondering if we should leave this flag for one more release but make it no-op with a warning that it's not used anymore and that it'll be removed in 2.23.

File: pkg/install/images/images.go [hunk 3]
```diff
@@ -67,47 +72,179 @@ import (
 
 const mockNamespaceName = "mock-namespace"
 
-func ExtractAddonsFromDockerImage(ctx context.Context, log logrus.FieldLogger, dockerBinary string, imageName string) (string, error) {
-	tempDir, err := os.MkdirTemp("", "imageloader*")
+type SourceDestImages struct {
```
Comment (xmudrii) [inline pkg/install/images/images.go]: This seems to be a single source-destination. Maybe we should call it something like `ImageSourceDest` (without `s`)?

**Comment by ahmedwaleedmalik** [comment]
/approve

It looks good to me. I think @xmudrii wanted to take a look as well.

**Review by xmudrii — COMMENTED** [review]
Some tiny nits. I'm very happy to see that we got crane working for us! :tada:

File: go.mod (hunk 1 above)
Comment (embik) [reply go.mod:355]: No, it was done by `go get`. I can change the file manually if you want, but this is how it ended up just with go tooling.

File: cmd/kubermatic-installer/cmd_mirror_images.go (hunk 2 above)
Comment (embik) [reply cmd/kubermatic-installer/cmd_mirror_images.go:107]: Sure, we can do that.

File: pkg/install/images/images.go (hunk 3 above)
Comment (embik) [reply pkg/install/images/images.go]: yeah that works. I wasn't too happy about the naming either.

File: go.mod (hunk 1 above)
Comment (xmudrii) [reply go.mod:355]: Then it's fine, leave it as it is.

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve

**Pull request merged**